js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetEdges.tsx

```diff
@@ -43,15 +51,33 @@ const AssetEdgeSet: React.FC<{
         <path d="M 0 0 L 8 5 L 0 10 z" fill={color} />
       </marker>
     </defs>
-    {edges.map((edge, idx) => (
-      <path
-        key={idx}
-        d={buildSVGPath({source: edge.from, target: edge.to})}
-        stroke={color}
-        strokeWidth={strokeWidth}
-        fill="none"
-        markerEnd={`url(#arrow${btoa(color)})`}
-      />
-    ))}
+    {edges
+      .filter(
+        (edge) =>
+          doesViewportContainPoint(edge.from, viewportRect) ||
+          doesViewportContainPoint(edge.to, viewportRect),
+      )
```

Comment:
Hmm I think we've explored this in the past but it causes flickering when the line spans all the way across the viewport.  If you have edge "A => B", where A and B are more than one full viewport size apart, as you try to pan to follow the edge it will disappear.  I think it'd be preferable (but ofc also slower) to render all lines whose bounding boxes intersect the viewport.

Reply:
Ahh, I see what you mean,